share/availability/light/availability.go

```diff
@@ -66,6 +64,22 @@ func (la *ShareAvailability) SharesAvailable(ctx context.Context, header *header
 		return nil
 	}
 
+	// Prevent multiple sampling sessions for the same header height
+	lockChan, loaded := la.activeHeights.LoadOrStore(header.Height(), make(chan struct{}))
+	if loaded {
+		select {
+		case <-lockChan.(chan struct{}):
```

Comment:
So the waiter sits here until the first caller is done and then goes over the same flow again. As the state is stored, the waiter sees that the sampling operation finished earlier and short circuits. In case the original operation failed, the waiter retries. Do we want to retry for each waiter out there?

Reply:
Yes, next job will find result of previous job from store. It will retry if there is still time within provided ctx deadline. it is good to keep trying if caller is waiting.

Reply:
In case there 3 waiters, doesn't that mean all of them will do the same retrying?

Reply:
@walldiss

Reply:
If all failed and all have time left to retry after previous one fail I see no issue. Availability supposed to succeed eventually.

Reply:
There was a bug causing second and further calls to start in parallel. Fixed in new commit

Reply:
Yes, so this is a bug